### PR TITLE
NetBSD fixes

### PR DESCRIPTION
**Pull request by 0-wiz-0**
This is a collection of the patches from pkgsrc to make libuv work better on NetBSD.

### CODE REVIEW

File: src/unix/thread.c [hunk 1]
```diff
@@ -708,7 +708,7 @@ int uv_cond_init(uv_cond_t* cond) {
   if (err)
     return UV__ERR(err);
 
-#if !(defined(__ANDROID_API__) && __ANDROID_API__ < 21)
+#if defined(HAVE_PTHREAD_CONDATTR_SETCLOCK)
```
Comment (saghul) [inline src/unix/thread.c]: This won't work for GYP or CMake builds.

File: src/unix/netbsd.c [hunk 2]
```diff
@@ -92,6 +93,22 @@ int uv_exepath(char* buffer, size_t* size) {
 
   /* Set new size. */
   *size = strlen(buffer);
+#else
+  ssize_t n;
```
Comment (saghul) [inline src/unix/netbsd.c]: don't we have this code somewhere else too? maybe we can DRY it?

File: src/unix/netbsd.c [hunk 3]
```diff
@@ -92,6 +93,22 @@ int uv_exepath(char* buffer, size_t* size) {
...
+
+  if (buffer == NULL || size == NULL || *size == 0)
+    return -EINVAL;
```
Comment (cjihrig) [inline src/unix/netbsd.c]:
```suggestion
    return UV_EINVAL;
```

File: src/unix/netbsd.c (hunk 2 above)
Comment (cjihrig) [reply src/unix/netbsd.c]: It's in `src/unix/procfs-exepath.c`.

**Comment by krytarowski** [comment]
Part of these patches are for NetBSD 6.x, do we still want to clobber 3rd party with EOL version of NetBSD?

File: src/unix/netbsd.c (hunk 2 above)
Comment (0-wiz-0) [reply src/unix/netbsd.c]: Suggestions what I should do here?

File: src/unix/netbsd.c (hunk 3 above)
Comment (0-wiz-0) [reply src/unix/netbsd.c]: Done, thanks.

**Comment by 0-wiz-0** [comment]
I've removed the NetBSD-6 specific patch.

File: src/unix/netbsd.c (hunk 2 above)
Comment (saghul) [reply src/unix/netbsd.c]: Use the function defined there as the fallback, instead of duplicating it here, if possible.

Comment (0-wiz-0) [reply src/unix/netbsd.c]: You know your build system best - what in particular do you recommend?

**Comment by 0-wiz-0** [comment]
I was waiting for comments on how to proceed with the exepath code.

**Comment by krytarowski** [comment]
https://github.com/libuv/libuv/pull/2507

**Comment by krytarowski** [comment]
I would drop patches for netbsd-7 here too.. 9.0 is planned to be releases soon and -7 will fade away officially.

File: src/unix/netbsd.c (hunk 2 above)
Comment (vtjnash) [reply src/unix/netbsd.c:97]: I think you could re-arrange that file into being header, so we can `#include` it on the necessary platforms.

**Comment by vtjnash** [comment]
Sorry that took so long to reply. This seems good overall, but needs a rebase now due to merge conflicts. Then happy to land for you once the code-duplication comment is addressed.

**Comment by 0-wiz-0** [comment]
Sorry, I lost interest in this, and it doesn't improve the 'make test' situation. I'll file a report about the broken tests.

**Pull request closed without merging**